src/rpc/blockchain.cpp

```diff
@@ -722,14 +723,30 @@ static RPCHelpMan getblock()
             throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Block not found");
         }
 
-        block = GetBlockChecked(chainman.m_blockman, pblockindex);
+        if (chainman.m_blockman.IsBlockPruned(pblockindex)) {
+            throw JSONRPCError(RPC_MISC_ERROR, "Block not available (pruned data)");
+        }
+
+        if (verbosity <= 0 && !RPCSerializationFlags()) {
+            // This one case doesn't need to parse the block at all
+            if (!node::ReadRawBlockFromDisk(raw_block, pblockindex->GetBlockPos(), Params().MessageStart())) {
```

Comment:
nit: Maybe move this into a GetRawBlockChecked to avoid the duplicate pruned check when `GetBlockChecked` below is called?